adapters/akka-http/src/main/scala/caliban/AkkaHttpAdapter.scala

```diff
@@ -107,14 +110,47 @@ trait AkkaHttpAdapter {
       }
     } ~
       post {
-        entity(as[GraphQLRequest])(
-          completeRequest(
-            interpreter,
-            skipValidation = skipValidation,
-            enableIntrospection = enableIntrospection,
-            contextWrapper = contextWrapper
-          )
-        )
+        extractRequest { request =>
+          parameters((Symbol("query").?, Symbol("operationName").?, Symbol("variables").?, Symbol("extensions").?)) {
+            case (query @ Some(_), op, vars, ext) =>
+              json
+                .parseHttpRequest(query, op, vars, ext)
+                .fold(
+                  failWith,
+                  completeRequest(
+                    interpreter,
+                    skipValidation = skipValidation,
+                    enableIntrospection = enableIntrospection,
+                    contextWrapper = contextWrapper
+                  )
+                )
+            case _ if request.entity.contentType.mediaType == `application/graphql` =>
+              onComplete(
```

Comment:
Could we maybe use `entity(as[String])` here?

Reply:
Haha, I had initially thought that using that would fail on an invalid json string since the post body would not be double quoted and have newlines escaped, and then in true programmer form, was too lazy to even give it a shot.  But of course, it does in fact work 😄 
I've updated the PR accordingly.